modules/live-schedule-impl/src/main/java/org/opencastproject/liveschedule/impl/LiveScheduleServiceImpl.java

```diff
@@ -450,29 +450,43 @@ void publishToSearch(MediaPackage mp) throws LiveScheduleException {
   }
 
   void retract(MediaPackage mp) throws LiveScheduleException {
+    Organization org = securityService.getOrganization();
+    User prevUser = org != null ? securityService.getUser() : null;
     try {
-      List<Job> jobs = new ArrayList<Job>();
+      securityService.setUser(SecurityUtil.createSystemUser(systemUserName, org));
       Set<String> elementIds = new HashSet<String>();
-      // Remove media package from the search index
       String mpId = mp.getIdentifier().toString();
       logger.info("Removing LIVE media package {} from the search index", mpId);
 
-      jobs.add(searchService.delete(mpId));
-      // Retract elements
       for (MediaPackageElement mpe : mp.getElements()) {
         if (!MediaPackageElement.Type.Publication.equals(mpe.getElementType())) {
           elementIds.add(mpe.getIdentifier());
         }
       }
-      jobs.add(downloadDistributionService.retract(CHANNEL_ID, mp, elementIds));
 
-      if (!waitForStatus(jobs.toArray(new Job[jobs.size()])).isSuccess()) {
-        throw new LiveScheduleException("Removing live media package from search did not complete successfully");
+      List<String> failedJobs = new ArrayList<>();
+      // Remove media package from the search index
+      Job searchDeleteJob = searchService.delete(mpId);
+      if (!waitForStatus(searchDeleteJob).isSuccess()) {
+        failedJobs.add("Search Index");
+      }
+
+      // Removing media from the download distribution service
+      Job distributionRetractJob =  downloadDistributionService.retract(CHANNEL_ID, mp, elementIds);
+      if (!waitForStatus(distributionRetractJob).isSuccess()) {
+        failedJobs.add("Distribution");
+      }
+
+      if (!failedJobs.isEmpty()) {
+        throw new LiveScheduleException(
+            String.format("Removing live media package %s from %s failed", mpId, String.join(" and ", failedJobs)));
```

Comment:
While debugging, I had some strange behavior. I am not 100% certain, but I think it could be some race condition when executing both jobs in parallel. So I made the jobs executing in series (they are not taking long anyway).